include/boost/json/detail/basic_parser.hpp

```diff
@@ -199,7 +198,8 @@ class basic_parser
     template<bool StackEmpty, class Handler>
     const char* parse_false(Handler& h, const char* p);
     
-    template<bool StackEmpty, bool AllowInvalid, class Handler>
+    template<bool StackEmpty, bool IsKey,
+        bool AllowInvalid, class Handler>
```

Comment:
I think we should rename `AllowInvalid` to `AllowBadUTF8`. This change should be in its own commit.